src/component/ResponsiveContainer.js

```diff
@@ -7,56 +7,70 @@ import pureRender from '../util/PureRender';
 import { getPercentValue, isPercent } from '../util/DataUtils';
 import { warn } from '../util/LogUtils';
 
-const render = ({ width, height, container, children }) => {
+const render = ({ aspect, width, height, minWidth, minHeight, container, children }) => {
+
   warn(isPercent(width) || isPercent(height),
-    `The width(%s) and height(%s) are both fixed number,
-     maybe you don't need to use ResponsiveContainer.`,
+    `The width(%s) and height(%s) are both fixed numbers,
+     maybe you don't need to use a ResponsiveContainer.`,
     width, height
   );
 
-  const calculatedWidth = getPercentValue(width, container.width);
-  const calculatedHeight = getPercentValue(height, container.height);
+  warn(!aspect || aspect > 0,
+       'The aspect(%s) must be greater than zero.',
+       aspect);
 
-  warn(calculatedWidth > 0 && calculatedHeight > 0,
-    `The width(%s) and height(%s) of chart should be greater than 0,
-    please check the style of container, or the props width(%s) and height(%s).`,
-     calculatedWidth, calculatedHeight, width, height
-  );
+  let calculatedWidth = getPercentValue(width, container.width);
```

Comment:
If I have set the width to be `40%`, then the `calculatedWidth` would be `16%` of the parent node because of the following calculation. 
```
let calculatedWidth = getPercentValue(width, container.width);
```